### PR TITLE
[CCIP-5718] pkg/reader: batch calls in GetChainFeePriceUpdate

### PR DESCRIPTION
**Pull request by makramkd**
Also added tests for various scenarios including happy path, empty selectors, batch call errors, partial successes, and missing results.

Integration tests enhanced here: https://github.com/smartcontractkit/chainlink/pull/17363

### CODE REVIEW

File: pkg/reader/ccip.go [hunk 1]
```diff
@@ -936,33 +936,119 @@ func (r *ccipChainReader) GetChainFeePriceUpdate(ctx context.Context, selectors
...
+}
+
+// processFeePriceUpdateResults iterates through batch results, validates them,
+// and populates the feeUpdates map.
+func (r *ccipChainReader) processFeePriceUpdateResults(
+	lggr logger.Logger,
+	selectors []cciptypes.ChainSelector,
+	results []types.BatchReadResult,
+	feeUpdates map[cciptypes.ChainSelector]cciptypes.TimestampedBig,
+) map[cciptypes.ChainSelector]cciptypes.TimestampedBig {
```
Comment (0xnogo) [inline pkg/reader/ccip.go:1012]: I think the map is mutated in place. If so, we could drop the return.

File: pkg/reader/ccip.go [hunk 2]
```diff
@@ -936,33 +936,119 @@ func (r *ccipChainReader) GetChainFeePriceUpdate(ctx context.Context, selectors
...
+			lggr.Warnw("failed to get chain fee price update from batch result",
+				"chain", chain,
+				"err", err)
 			continue
 		}
+
+		// Type assert the result
+		update, ok := val.(*cciptypes.TimestampedUnixBig)
+		if !ok || update == nil {
+			lggr.Errorw("Invalid type or nil value received for chain fee price update",
```
Comment (0xnogo) [inline pkg/reader/ccip.go:1035]: I'd rather go with a `Warnw` as the function is dealing gracefully with that.

File: pkg/reader/ccip.go [hunk 3]
```diff
@@ -936,33 +936,119 @@ func (r *ccipChainReader) GetChainFeePriceUpdate(ctx context.Context, selectors
...
+	// 3. Find FeeQuoter Results
+	var feeQuoterResults []types.BatchReadResult
+	found := false
+	for contract, results := range batchResult {
+		if contract.Name == consts.ContractNameFeeQuoter {
+			feeQuoterResults = results
+			found = true
+			break // Found the results, exit loop
+		}
+	}
```
Comment (0xnogo) [inline pkg/reader/ccip.go:987]: Can we do something like this instead:

```
feeQuoterResults, found := batchResult[types.BoundContract{
    Name: consts.ContractNameFeeQuoter,
}]
if !found {
    lggr.Errorw("FeeQuoter results missing from batch response")
    return feeUpdates
}
```

**Review by 0xnogo — COMMENTED** [review]
Minor comments

File: pkg/reader/ccip.go (hunk 3 above)
Comment (makramkd) [reply pkg/reader/ccip.go:987]: I don't think so cuz `BoundContract` has an address as well?

File: pkg/reader/ccip.go (hunk 1 above)
Comment (makramkd) [reply pkg/reader/ccip.go:1012]: Ah nice find, don't like that actually, will just init the map and return it.
